### ISSUE TITLE
Multi-layered feature metadata?

### ISSUE THREAD

**Issue by eitsupi** [devcontainers/cli#200]
In https://github.com/devcontainers/images, it seems that the builds are performed with the settings for vscode written in `devcontainer.json`.
How is this intended to work?
So far, when I do a build on hand based on these images, the settings for vscode defined in https://github.com/devcontainers/images do not seem to be reflected in the container on hand, is this the intended behavior?
Perhaps I am missing a spec, but I am wondering which settings up to which are normal to be reflected in the container at hand.

In other words, while it is often convenient to have the settings for vscode defined in upstream reflected at hand, as it eliminates the need to write most of the devcontainer.json at hand.
But without the option to remove metadata defined by upstream, we may suffer from unnecessary settings defined upstream.
(Similar to user frustration with not being able to remove `EXPOSE` or `VOLUME` from upstream Dockerfiles)

**Comment by chrmarti** [comment]
You can override the settings in your devcontainer.json at:
```jsonc
{
	// ...
	"customizations": {
		"vscode": {
			"settings": {
				// HERE
			}
		}
	}
}
```

**Comment by chrmarti** [comment]
This spec proposal has additional details: https://github.com/devcontainers/spec/blob/main/proposals/image-metadata.md

**Comment by eitsupi** [comment]
The templates in https://github.com/devcontainers/templates at the moment don't seem to hold settings for vscode (unlike before), are you assuming this feature will be added?

I am currently working on creating templates and am wondering if the settings for vscode should be included in the template.
(If this feature is implemented, I will not need to include the settings for vscode in the templates since all vscode settings are written at image build time)

**Comment by chrmarti** [comment]
The https://github.com/devcontainers/templates are using our base images which come with settings included, e.g.: https://github.com/devcontainers/images/blob/main/src/javascript-node/.devcontainer/devcontainer.json.

**Comment by eitsupi** [comment]
Yeah, I created a template with the same strategy and it seems to be working well.
https://github.com/rocker-org/devcontainer-templates/blob/6c46be9b038d42cd4c54f1f0646dfca13b3a60e3/src/r-ver/.devcontainer/devcontainer.json

Perhaps my version of VSCode Dev Container at the time was older and not working well.
